gtk4/src/subclass/widget.rs

```diff
@@ -895,6 +896,118 @@ pub unsafe trait WidgetClassSubclassExt: ClassStruct {
         }
     }
 
+    /*
+    fn install_action<F: Fn(&Widget, &str, Option<&glib::Variant>) + 'static>(
+        &mut self,
+        action_name: &str,
+        parameter_type: Option<&str>,
+        f: F,
+    ) {
+        unsafe {
+            unsafe extern "C" fn activate_trampoline<
+                F: Fn(&Widget, &str, Option<&glib::Variant>) + 'static,
+            >(
+                this: *mut ffi::GtkWidget,
+                action: *const libc::c_char,
+                parameter: *mut glib::ffi::GVariant,
+            ) {
+                f(
+                    &from_glib_borrow(this),
+                    glib::GString::from_glib_borrow(action).as_str(),
+                    Option::<glib::Variant>::from_glib_borrow(parameter)
+                        .as_ref()
+                        .as_ref(),
+                )
+            }
+
+            let widget_class = self as *mut _ as *mut ffi::GtkWidgetClass;
+            let activate_func = Some(activate_trampoline::<F> as _);
+
+            ffi::gtk_widget_class_install_action(
+                widget_class,
+                action_name.to_glib_none().0,
+                parameter_type.to_glib_none().0,
+                activate_func,
+            );
+        }
+    }
+
+
+    fn query_action(
+        &self,
+        index: u32,
+    ) -> Option<(glib::Type, glib::GString, Option<glib::VariantType>, glib::GString)> {
```

Comment:
`property_name` (the last string) can be `NULL` according to the code. `parameter_type` (the `VariantType`) maybe not.

This function needs better documentation on the C side and annotations.

Also this function is basically an iterator (you might want to add one)

Reply:
I have disabled this as well as I don't really need it for now

Reply:
Can you create issues for that? On gtk (annotations) and here?

Reply:
This one is still pending an issue or two :)

Reply:
The first issue is #111, the upstream one https://gitlab.gnome.org/GNOME/gtk/-/issues/3757